lib/libvfio-user.c

```diff
@@ -524,25 +524,57 @@ handle_dma_unmap(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
                  struct vfio_user_dma_region *region)
 {
     int ret;
+    char *bitmap;
     char rstr[1024];
 
     assert(vfu_ctx != NULL);
     assert(msg != NULL);
     assert(region != NULL);
 
-    if (msg->in_size != sizeof(*region)) {
+    if (msg->in_size < sizeof(*region)) {
         vfu_log(vfu_ctx, LOG_ERR, "bad size of DMA unmap region %zu",
                 msg->in_size);
         return ERROR_INT(EINVAL);
     }
 
-    snprintf(rstr, sizeof(rstr), "[%#lx, %#lx) offset=%#lx "
-            "prot=%#x flags=%#x", region->addr, region->addr + region->size,
-            region->offset, region->prot, region->flags);
+    snprintf(rstr, sizeof(rstr), "[%#lx, %#lx) offset=%#lx prot=%#x flags=%#x",
+             region->addr, region->addr + region->size,
+             region->offset, region->prot, region->flags);
 
     vfu_log(vfu_ctx, LOG_DEBUG, "removing DMA region %s", rstr);
 
-    if (region->flags != 0) {
+    if (region->flags == VFIO_DMA_UNMAP_FLAG_GET_DIRTY_BITMAP) {
+        if (msg->in_size != sizeof(struct vfio_user_dma_region) + sizeof(struct vfio_user_bitmap)) {
```

Comment:
nit, use sizeof(*region) like :534
Also, this should be msg->in_size < not != too

Reply:
ok